src/bitmap/mutable.rs

```diff
@@ -309,23 +310,92 @@ impl FromIterator<bool> for MutableBitmap {
     }
 }
 
+// [7, 6, 5, 4, 3, 2, 1, 0], [15, 14, 13, 12, 11, 10, 9, 8]
+// [00000001_00000000_00000000_00000000_...] // u64
+/// # Safety
+/// The iterator must be trustedLen and its len must be least `len`.
 #[inline]
-fn extend<I: Iterator<Item = bool>>(buffer: &mut [u8], length: usize, mut iterator: I) {
-    let chunks = length / 8;
-    let reminder = length % 8;
-
-    buffer[..chunks].iter_mut().for_each(|byte| {
-        (0..8).for_each(|i| {
-            *byte = set(*byte, i, iterator.next().unwrap());
-        })
-    });
-
-    if reminder != 0 {
-        let last = &mut buffer[chunks];
-        iterator.enumerate().for_each(|(i, value)| {
-            *last = set(*last, i, value);
-        });
+unsafe fn get_chunk_unchecked(iterator: &mut impl Iterator<Item = bool>) -> u64 {
+    let mut byte = 0u64;
+    let mut mask;
+    for i in 0..8 {
+        mask = 1u64 << (8 * i);
+        for _ in 0..8 {
+            let value = match iterator.next() {
+                Some(value) => value,
+                None => unsafe { unreachable_unchecked() },
+            };
+
+            byte |= match value {
+                true => mask,
+                false => 0,
+            };
+            mask <<= 1;
+        }
+    }
+    byte
+}
+
+/// # Safety
+/// The iterator must be trustedLen and its len must be least `len`.
+#[inline]
+unsafe fn get_byte_unchecked(len: usize, iterator: &mut impl Iterator<Item = bool>) -> u8 {
+    let mut byte_accum: u8 = 0;
+    let mut mask: u8 = 1;
+    for _ in 0..len {
```

Comment:
Wondering if this could be improved by chunking the iterator, and doing something like:

```rust
          chunk.iter()
                .enumerate()
                .for_each(|(i, b)| {
                    *byte |= if b { 1 << i } else { 0 };
                });
```

At least for the binary comparison code, this was leading to great, unrolled, code.

Reply:
isn't chunking only available in slices, e.g. `&[bool].chunk_exact` (or in bitmaps, via `u8` chunks)?

Reply:
Yeah you are right. There might be some remaining optimizations possible, let me take a look :)